### ISSUE TITLE
CI is broken on Bazel Rules

### ISSUE THREAD

**Issue by aiuto** [bazelbuild/examples#240]
I think because of 87a4812cb23f7e7969d74cc073579fb82540c0f6

**Comment by meteorcloudy** [comment]
This is broken by an incompatible change in Bazel: https://github.com/bazelbuild/bazel/issues/14361

https://github.com/bazelbuild/examples/commit/87a4812cb23f7e7969d74cc073579fb82540c0f6 tried to ping the Bazel version, but since the latest release version already contain the breaking change, the main pipeline is also broken now.

**Comment by meteorcloudy** [comment]
Is this examples repo still maintained? Who should own the broken examples now after the original authors left?

FYI sdtwigg

**Comment by aiuto** [comment]
cc: @philomathing @jdob @fweikert @gregestren 

examples should be maintained. We need a place to hold complete examples of complex things. The documentation can point out to it.

@philomathing, @jdob, @fweikert, and I talked about examples and keeping them running last week. We saw a few requirements.

Examples will eventually need versioning of some sort. An example may work with 4.x but break with 5.x, so we need to ensure a few things
- that the user can clearly see what version of Bazel the example is for. We decided to do that with a folder naming convention rather than branches for each LTs. TBD: specify the convention
- that the example author can easily specify which version of Bazel to use. Perhaps that is a .bazelversion in each example. Maybe we pick it up by path to the example.
- that the CI chooses the bazel version that the author intended. Right now we put the bazel version in the .bazelci file. That is slightly error prone because that means CI and interactive use different mechanisms to pick the version.

Some team must be accountable for keeping examples running.
- If Bazel at head break examples, it is probably the responsibility of the Bazel team to fix the example. That could be by updating it, or by cloning it to a version for the old LTS release and for the next one.
- But it would be better if we could farm some of that work to the example author. We did decide we want to have an explicit link from each example to the author. Maybe we can use that to notify them of breakages.

Since I am a little ahead of the learning curve on how our CI runs for examples than the docs team, I may take a try at fixing the immediate problem now and letting others improve things over time. Towards that end, my plan is to first send a PR to disable the broken tests. This will get CI green and show @jdob how we set up the run for each test.  After that, we can figure out the naming scheme for LTS specific versions and split the broken ones.
